checkov/common/images/image_referencer.py

```diff
@@ -75,3 +88,149 @@ def inspect(image_name: str) -> str:
+        if runner_filter.checks and all(not check.startswith("CKV_CVE") for check in runner_filter.checks):
+            return None
+
+        resources = [
+            block
+            for node, block in graph_connector.nodes(data=True)
+            if block and block.get(CustomAttributes.RESOURCE_TYPE) in supported_resource_types
+        ]
+        if not resources:
+            return None
+
+        images = self.extract_images(resources=resources)
+        if not images:
+            return None
+
+        logging.info(f"Found {len(images)} image references {[image.name for image in images]}")
+
+        report = Report(CheckType.SCA_IMAGE)
+        root_path = Path(root_path) if root_path else None
+        check_class = f"{image_scanner.__module__}.{image_scanner.__class__.__qualname__}"
+        report_type = CheckType.SCA_IMAGE
+
+        for image in images:
+            self.add_image_record(
+                report=report,
+                root_path=root_path,
+                check_class=check_class,
+                dockerfile_path=image.file_path,
+                image=image,
+                runner_filter=runner_filter,
+                report_type=report_type,
+            )
+
+        return report
+
+    def add_image_record(
+        self,
+        report: Report,
+        root_path: Path | None,
+        check_class: str,
+        dockerfile_path: str,
+        image: Image,
+        runner_filter: RunnerFilter,
+        report_type: str,
+    ) -> None:
+        """Adds an image record to the given report, if possible"""
+
+        cached_results: dict[str, Any] = image_scanner.get_scan_results_from_cache(f"image:{image.name}")
+        if cached_results:
+            logging.info(f"Found cached scan results of image {image.name}")
+
+            result = cached_results.get("results", [{}])[0]
+            vulnerabilities = result.get("vulnerabilities") or []
+            image_id = self.extract_image_short_id(result)
+            image_details = self.get_image_details_from_twistcli_result(scan_result=result, image_id=image_id)
+            if root_path:
+                try:
+                    dockerfile_path = str(Path(dockerfile_path).relative_to(root_path))
+                except ValueError:
+                    # Path.is_relative_to() was implemented in Python 3.9
+                    pass
+            rootless_file_path = dockerfile_path.replace(Path(dockerfile_path).anchor, "", 1)
+
+            parse_vulns_to_records(
+                report=report,
+                check_class=check_class,
+                scanned_file_path=os.path.abspath(dockerfile_path),
+                rootless_file_path=f"{rootless_file_path} ({image.name} lines:{image.start_line}-{image.end_line} ({image_id}))",
+                runner_filter=runner_filter,
+                vulnerabilities=vulnerabilities,
+                packages=[],
+                license_statuses=[],
```

Comment:
make sure to use both packages and license_statuses, as it happen now in sca_image runner
in addition, please try to use the existing code from there as much as possible. I think it can reduce a lot of code in the rest of the method

Reply:
the duplication is intentional, because I want to adjust the old code to work like this approach. I also wanted to reduce the amount of changes in this PR, so it will be done with a follow up PR 🚀